xilem_web: Fix clippy::multiple_bound_locations lints.

waywardmonkeys opened the pull request:
Define generic bounds once rather than partly in the bound predicate and partly in the `where` clause.

Philipp-M reviewed (approved):
Not that I have strong feelings about this, but I've put the bounds in the where clause already in #197 ([here](https://github.com/linebender/xilem/blob/b3f1068299d3941e9e4575c489c8ae299a519250/crates/xilem_web/src/vecmap.rs)), I think having them in one place (`where ...`) would be better?

waywardmonkeys commented:
I tired that! But apparently `?Sized` couldn’t be there … will repro later and make sure I didn’t do something dumb.

waywardmonkeys commented:
@Philipp-M  So, I'd done something dumb earlier when I got an error. I've updated the PR per your suggestion (and added one case that the lint didn't catch for the `impl` block).
